core/src/network/handler/send.rs

```diff
@@ -0,0 +1,88 @@
+use crate::{
+    action::{Action, ActionWrapper},
+    context::Context,
+    instance::dispatch_action,
+    network::direct_message::DirectMessage,
+    workflows::respond_validation_package_request::respond_validation_package_request,
+};
+use futures::executor::block_on;
+use holochain_core_types::cas::content::Address;
+use std::{sync::Arc, thread};
+
+use holochain_net_connection::protocol_wrapper::MessageData;
+
+fn log<T: Into<String>>(context: &Arc<Context>, msg: T) {
+    context.logger.lock().unwrap().log(msg.into());
+}
+
+/// We got a ProtocolWrapper::SendMessage, this means somebody initiates message roundtrip
+/// -> we are being called
+pub fn handle_send(message_data: MessageData, context: Arc<Context>) {
+    let message: DirectMessage =
+        serde_json::from_str(&serde_json::to_string(&message_data.data).unwrap()).unwrap();
+
+    match message {
+        DirectMessage::Custom(_) => log(&context, "DirectMessage::Custom not implemented"),
+        DirectMessage::RequestValidationPackage(address) => {
+            // Async functions only get executed when they are polled.
+            // I don't want to wait for this workflow to finish here as it would block the
+            // network thread, so I use block_on to poll the async function but do that in
+            // another thread:
+            thread::spawn(move || {
+                block_on(respond_validation_package_request(
```

Comment:
Agree, we don't want to block the network thread, but we also want to be careful not to duplicate the pattern of spawning hard os threads too much. Isn't that the point of thread pool executors in futures?

Reply:
Well yeah, we could (and maybe should soon) implement our own executor for that. With the old futures I could have just called that function and not wait for the future result. But with async/await they decided for not executing the async code at all until somebody starts polling.. Would leave it like this for now and add a custom global future executor soon.

Reply:
:+1: